apps/examples/testcase/ta_tc/task_manager/itc/itc_taskmanager_main.c

```diff
@@ -247,108 +336,164 @@ void cleanup_task5(void)
-	TC_ASSERT_EQ_CLEANUP("task_manager_unregister", ret, OK, cleanup_task5());
+	if (ret != OK) {
+		printf("ERROR: failed task_manager_unregister in taskmanager_task3 ERR: %d\n", ret);
+		cleanup_task5();
+		return;
+	}
 
 }
 
-int itc_taskmanager_task3_main(int argc, char *argv[])
+int taskmanager_task3_main(int argc, char *argv[])
 {
-	itc_taskmanager_task3();
+	taskmanager_task3();
 	return 0;
 }
 
-void itc_taskmanager_task4(void)
+void taskmanager_task4(void)
 {
 	int tm_sample_handle4;
+	int ret;
 
 	tm_sample_handle4 = task_manager_register_task("task5", 100, 1024, not_builtin_task_1, NULL, TM_APP_PERMISSION_ALL, TM_RESPONSE_WAIT_INF);
-	TC_ASSERT_GEQ("task_manager_register_task", tm_sample_handle4, 0);
+	if (tm_sample_handle4 < 0) {
+		printf("ERROR: failed task_manager_register_task in taskmanager_task4 ERR: %d\n", tm_sample_handle4);
+		return;
+	}
 
 	unregister_task5_flag = false;
-	int ret;
 
 	ret = task_manager_start(tm_sample_handle4, TM_RESPONSE_WAIT_INF);
-	TC_ASSERT_EQ("task_manager_start", ret, OK);
+	if (ret != OK) {
+		printf("ERROR: failed task_manager_start in taskmanager_task4 ERR: %d\n", ret);
+		unregister_tasks_3(tm_sample_handle4);
+		return;
+	}
 
 	while (1) {
 		usleep(1);
 		if (unregister_task5_flag) {
 			ret = task_manager_unregister(tm_sample_handle4, TM_RESPONSE_WAIT_INF);
-			TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+			if (ret != OK) {
+				printf("ERROR: failed task_manager_unregister in taskmanager_task4 ERR: %d\n", ret);
+			}
 		}
 	}
 }
 
-int itc_taskmanager_task4_main(int argc, char *argv[])
+int taskmanager_task4_main(int argc, char *argv[])
 {
-	itc_taskmanager_task4();
+	pid_tm_task4 = getpid();
+
+	taskmanager_task4();
 	return 0;
 }
 
 void unregister_tasks_2(void)
 {
 	int ret;
-
 	int tm_sample_handle2;
-
 	tm_appinfo_list_t *sample_list_info;
+
 	if (unregister_task5_flag) {
 		sample_list_info = ((tm_appinfo_list_t *)task_manager_getinfo_with_name("task5", TM_RESPONSE_WAIT_INF));
-		TC_ASSERT_NEQ("task_manager_getinfo_with_name", sample_list_info, NULL);
+		if (sample_list_info != NULL) {
+			printf("ERROR: failed task_manager_getinfo_with_name in unregister_tasks_2\n");
+		}
 
 		tm_sample_handle2 = sample_list_info->task.handle;
 		task_manager_clean_infolist(&sample_list_info);
-		TC_ASSERT_NEQ("task_manager_clean_infolist", (tm_appinfo_list_t *)&sample_list_info, NULL);
+		if ((tm_appinfo_list_t *)&sample_list_info == NULL) {
+			printf("ERROR: failed task_manager_clean_infolist in unregister_tasks_2\n");
+		}
 
 		ret = task_manager_unregister(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
-		TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+		if (ret != OK) {
+			printf("ERROR: failed task_manager_unregister in unregister_tasks_2 ERR: %d\n", ret);
+		}
 	}
 
 	sample_list_info = ((tm_appinfo_list_t *)task_manager_getinfo_with_name("task4", TM_RESPONSE_WAIT_INF));
-	TC_ASSERT_NEQ("task_manager_getinfo_with_name", sample_list_info, NULL);
+	if (sample_list_info != NULL) {
```

Comment:
Please check the condition : if (sample_list_info != NULL) , i think it should be "sample_list_info == NULL"

Reply:
Please check similar case also.

Reply:
The given comment is being fixed.